doc/introduction/mid_circuit_measurements.rst

```diff
@@ -0,0 +1,456 @@
++--------------------------+-------------------------------------------+-----------------------------------------------------------+-------------------------------------------+----------------------------------+
+
+:math:`{}^1` In principle, parameter-shift differentiation is supported as long as no
+postselection is used. Parameters within conditionally applied operations will
+fall back to finite differences, so a proper value for ``h`` should be provided (see
+:func:`~.pennylane.gradients.finite_diff`).
+
+The strengths and weaknesses of the simulation techniques differ strongly and the best
+technique will depend on details of the simulation workflow. As a rule of thumb:
+
+- dynamic one-shot sampling excels in the many-measurements-few-shots regime,
+
+- the tree-traversal technique can handle large-scale simulations with many shots
+  and measurements, and
+
+- deferred measurements are the generalist solution that enables mid-circuit measurement
+  support under (almost) all circumstances, but at large memory cost. It is the only method
+  supporting analytic simulations.
+
+By default, ``QNode``\ s use deferred measurements and dynamic one-shot sampling (if supported)
+when executed without and with shots, respectively. The method can be configured with
+the keyword argument ``mcm_method`` at ``QNode`` creation
+(see :ref:`"Configuring mid-circuit measurements" <mcm_config>`).
+
+.. _deferred_measurements:
+
+Deferred measurements
+*********************
+
+A quantum function with mid-circuit measurements can be executed via the
+`deferred measurement principle <https://en.wikipedia.org/wiki/Deferred_Measurement_Principle>`__.
+In PennyLane, this technique is available via ``mcm_method="deferred"`` or as the
+transform :func:`~.pennylane.defer_measurements`.
+
+The deferred measurement principle provides a powerful method to simulate
+mid-circuit measurements, conditional operations and measurement statistics
+in a differentiable and device-independent way. It adds an auxiliary qubit
+to the circuit for each mid-circuit measurement, leading to overheads of both
+memory and simulation time that scale exponentially with the number of measurements.
+
+.. code-block:: pycon
+
+    >>> deferred_qnode = qml.defer_measurements(my_qnode)
+    >>> pars = np.array([0.643, 0.246])
+    >>> deferred_qnode(*pars)
+    (tensor([0.90165331, 0.09834669], requires_grad=True),
+     tensor(0.09984972, requires_grad=True))
+
+The effect of deferring measurements becomes clear if we draw the ``QNode``
+before and after applying the transform:
+
+.. code-block:: pycon
+
+    >>> qml.draw(my_qnode)(*pars)
+    0: ──RY(0.64)─╭●───────RY(0.25)─┤  Probs
+    1: ───────────╰X──┤↗├──║────────┤
+                       ╚═══╩════════╡  <MCM>
+    >>> qml.draw(deferred_qnode)(*pars)
+    0: ──RY(0.64)─╭●────╭RY(0.25)─┤  Probs
+    1: ───────────╰X─╭●─│─────────┤
+    2: ──────────────╰X─╰●────────┤  <None>
```

Comment:
I'm a big fan of the specific example you've used as it perfectly shows in the drawing how we replace classical wires with quantum wires.